os/pm/pm_changestate.c

```diff
@@ -122,9 +124,11 @@ static void pm_timer(int domain, enum pm_state_e newstate)
 		int delay = pmtick[pdom->state] + pdom->btime - clock_systimer();
 		int left  = wd_gettime(pdom->wdog);
 		lldbg("\n[%s] - %d, delay = %d, left = %d, pdom->btime = %8lld, pdom->stime = %8lld, clock_systimer = %8lld\n",__FUNCTION__,__LINE__, delay, left, pdom->btime, pdom->stime, clock_systimer());
+		/* [AMOGH HASSIJA] Can you explain the reason for adding the 'if' conditional below? */
```

Comment:
Hi Amogh,
I synced this code from nuttx codebase, you may check https://github.com/apache/nuttx/blob/master/drivers/power/pm/activity_governor.c#L577.
I think it is to prevent from cases where the clock_systimer() is ahead of pdom->btime causing negative value.